pavelib/paver_tests/test_js_test.py

```diff
@@ -0,0 +1,139 @@
+"""Unit tests for the Paver JavaScript testing tasks."""
+
+import ddt
+from mock import patch
+from paver.easy import call_task
+
+import pavelib.js_test
+from .utils import PaverTestCase
+
+
+@ddt.ddt
+class TestPaverJavaScriptTestTasks(PaverTestCase):
+    """
+    Test the Paver JavaScript testing tasks.
+    """
+
+    EXPECTED_DELETE_JAVASCRIPT_REPORT_COMMAND = u'find {platform_root}/reports/javascript -type f -delete'
+    EXPECTED_INSTALL_NPM_ASSETS_COMMAND = u'install npm_assets'
+    EXPECTED_COFFEE_COMMAND = (
+        u'node_modules/.bin/coffee --compile `find {platform_root}/lms {platform_root}/cms '
+        u'{platform_root}/common -type f -name "*.coffee"`'
+    )
+    EXPECTED_JS_TEST_TOOL_OPTIONS = (
+        u"{platform_root}/lms/static/js_test.yml "
+        u"{platform_root}/lms/static/js_test_coffee.yml "
+        u"{platform_root}/cms/static/js_test.yml "
+        u"{platform_root}/cms/static/js_test_squire.yml "
+        u"{platform_root}/common/lib/xmodule/xmodule/js/js_test.yml "
+        u"{platform_root}/common/static/js_test.yml "
+        u"{platform_root}/common/static/js_test_requirejs.yml "
+        u"--use-firefox "
+        u"--timeout-sec 600 "
+        u"--xunit-report "
+        u"{platform_root}/reports/javascript/javascript_xunit.xml"
+    )
+    EXPECTED_COVERAGE_OPTIONS = (
+        u' --coverage-xml {platform_root}/reports/javascript/coverage.xml'
+    )
+
+    EXPECTED_COMMANDS = [
+        u"make report_dir",
+        u'git clean -fqdx test_root/logs test_root/data test_root/staticfiles test_root/uploads',
+        u"find . -name '.git' -prune -o -name '*.pyc' -exec rm {} \\;",
+        u'rm -rf test_root/log/auto_screenshots/*',
+        u"rm -rf /tmp/mako_[cl]ms",
+    ]
+
+    def setUp(self):
+        super(TestPaverJavaScriptTestTasks, self).setUp()
+
+        # Mock the paver @needs decorator
+        self._mock_paver_needs = patch.object(pavelib.js_test.test_js, 'needs').start()
+        self._mock_paver_needs.return_value = 0
+
+        # Mock shell commands
+        patcher = patch('pavelib.quality.sh')
```

Comment:
I think this is a copy/paste error. You might actually be calling sh in these tests at this point.